smartsim/experiment.py

```diff
@@ -827,6 +827,57 @@ def reconnect_orchestrator(self, checkpoint: str) -> Orchestrator:
             logger.error(e)
             raise
 
+    def preview(
+        self,
+        *args: t.Any,
+        output_format: previewrenderer.Format = previewrenderer.Format.PLAINTEXT,
+        verbosity_level: previewrenderer.Verbosity = previewrenderer.Verbosity.INFO,
+        output_filename: t.Optional[str] = None,
+    ) -> None:
+        """Preview entity information prior to launch. This method
+        aggregates multiple pieces of information to give users insight
+        into what and how entities will be launched.  Any instance of
+        ``Model``, ``Ensemble``, or ``Orchestrator`` created by the
+        Experiment can be passed as an argument to the preview method.
+        :param output_filename: Specify name of file and extension to write
+        preview data to. If no output filename is set, the preview will be
+        output to stdout. Defaults to None.
+        :type output_filename: str
+        :param output_format: Set output format. The possible accepted
+        output formats are `json`, `xml`, `html`, `plain_text`, `color_text`.
+        Defaults to 'plain_text'.
+        :type output_type: str
+        :param verbosity_level: Specify the verbosity level:
+            info: Display User defined fields and entities
+            debug: Display user defined field and entities and auto generated
+            fields.
+            developer: Display user defined field and entities, auto generated
+            fields, and run commands.
+            Defaults to info.
+        :type verbosity_level: str
+        """
+
+        # Retrive any active db jobs
+        active_dbjobs = self._control.active_orch_dict
+
+        preview_manifest = Manifest(*args)
+
+        rendered_preview = previewrenderer.render(
```

Comment:
I think this interface could be cleaned up a little bit.  It seems awkward that the ``Experiment`` API is passing what could be an intermediate representation between ``previewrenderer`` functions.  As the format types increase, it's going to become more difficult to keep track of proper configuration/types/representations between the functions.  It might be cleaner to do:

```python
if output_filename:
    previewrendered.render_to_file(self,
            preview_manifest,
            verbosity_level,
            output_format,
            output_filename,
            active_dbjobs,
    )
else:
    rendered_preview = previewrenderer.render_to_string(self,
            preview_manifest,
            verbosity_level,
            output_format,
            active_dbjobs,
    )
    logger.info(rendered_preview)
```

Reply:
Changed interface to example.